core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -2257,4 +2258,16 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
   @LibraryOperator(libraries = {SPARK})
   public static final SqlFunction GETBIT =
       BIT_GET.withName("GETBIT");
+
+  /** The RANDOM() function. Equivalent to RAND(). */
+  @LibraryOperator(libraries = {POSTGRESQL})
+  public static final SqlFunction RANDOM = new SqlRandFunction() {
```

Comment:
what's to keep you from doing RAND.withName("RANDOM").withReturnTypeInferance(...)` I think that's better than these overrides

Reply:
I'd prefer that as well. Those methods are part of SqlBasicFunction, but SqlRandFunction derives directly from SqlFunction.

Reply:
Done

Reply:
can't you make SqlRandFunction extend SqlBasicfunction

Reply:
SqlBasicFunction has a private constructor.
I'll look into making RAND be an instance of SqlBasicFunction then RANDOM being a variant instead.

Reply:
Done.

Reply:
the constructor is private because you're meant to use `.create()`

Reply:
Yes, I wound up adding more with*() functions to SqlBasicFunction for setting dynamic and the OperandTypeValidator and made both RAND() and RANDOM() instances of it.